scripts/west_commands/run_common.py

```diff
@@ -224,35 +235,46 @@ def do_run_common(command, user_args, user_runner_args, domain_file=None):
             # single-use commands in a dictionary so that they get executed
             # once per unique board name.
             if cache['BOARD_DIR'] not in processed_boards and 'SOC_FULL_DIR' in cache:
-                soc_yaml_file = Path(cache['SOC_FULL_DIR']) / 'soc.yml'
-                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
-                group_type = 'boards'
+                highest_priority = IGNORED_RUN_ONCE_PRIORITY
+                highest_entry = None
+                check_files = []
 
-                # Search for flash runner configuration, board takes priority over SoC
-                try:
-                    with open(board_yaml_file, 'r') as f:
-                        data_yaml = yaml.safe_load(f.read())
+                for directory in re.split(';', cache['SOC_DIRECTORIES']):
```

Comment:
Consider using the existing `CMakeCache` class method:
```suggestion
                for directory in cache.get_list('SOC_DIRECTORIES'):
```

Reply:
Fixed